File: model/Partners.js
const { DataTypes } = require('sequelize');
const db = require('../database');
const Department = require('./DepartmentModel');

const Partners = db.define('Partners', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    name: {
        type: DataTypes.STRING(255)
    },
    countries: {
        type: DataTypes.STRING(255)
    },
    open_scolars: {
        type: DataTypes.INTEGER
    },
    partners_photos: {
        type: DataTypes.BLOB('long')
    },
    level: {
        type: DataTypes.ENUM('Bachelor', 'Master')
    },
    semester: {
        type: DataTypes.ENUM('Winter', 'Summer')
    },
    dep_id: {
        type: DataTypes.INTEGER,
        references: {
            model: Department,
            key: 'dep_id'
        }
    }
}, {
    timestamps: false,
    freezeTableName: true,
    tableName: 'Partners',
    underscored: true
});

Partners.belongsTo(Department, { foreignKey: 'dep_id' });

Partners.sync({ force: false }).then(() => {
    console.log('Partners table synced');
}).catch(err => {
    console.error('Error syncing Partners table:', err);
});

module.exports = Partners;

File: java-service/src/main/java/com/uus/model/Partners.java
package com.uus.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "Partners")
public class Partners {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(length = 255)
    private String name;

    @Column(length = 255)
    private String countries;

    @Column(name = "open_scolars")
    private Integer open_scolars;

    @Lob
    @Column(name = "partners_photos", columnDefinition = "LONGBLOB")
    private byte[] partners_photos;

    @Column(length = 8)
    @Enumerated(EnumType.STRING)
    private Level level;

    @Column(length = 6)
    @Enumerated(EnumType.STRING)
    private Season semester;

    @Column(name = "dep_id")
    private Integer dep_id;

    public enum Season {
        Winter, Summer
    }
    
    public enum Level {
        Bachelor, Master
    }
}
